### PR TITLE
Adds missing "mode" to tasks with copy and file modules

### PR DESCRIPTION
**Pull request by frenzyfriday**
As a pull request owner and reviewers, I checked that:
- [X] Appropriate testing is done and actually running
- [X] Appropriate documentation exists and/or is up-to-date:
  - [X] README in the role
  - [X] Content of the docs/source is reflecting the changes

### CODE REVIEW

**Review by lewisdenny — CHANGES_REQUESTED** [review]
Looks good @frenzyfriday, thank you!

Are you going to enable `risky-file-permissions` lint so we can see if this now passes CI on this patch? [1]

[1] https://github.com/openstack-k8s-operators/ci-framework/blob/8cd2a2db84b2c89af8f854948b620f9462065501/.ansible-lint#L43C5-L43C27

File: playbooks/99-logs.yml [hunk 1]
```diff
@@ -80,6 +81,7 @@
       block:
         - name: Copy facts to dated directory
           ansible.builtin.copy:
+            mode: "0666"
```
[the remark this answers is not in the document]
Comment (frenzyfriday) [reply playbooks/99-logs.yml]: ack, updated

File: roles/reproducer/tasks/ci_job.yml [hunk 2]
```diff
@@ -25,6 +25,7 @@
           tags:
             - bootstrap
           ansible.builtin.copy:
+            mode: "0644"
```
[the remark this answers is not in the document]
Comment (frenzyfriday) [reply roles/reproducer/tasks/ci_job.yml]: ack, updated

**Comment by lewisdenny** [comment]
rebased to check current status

**Comment by lewisdenny** [comment]
recheck

File: roles/virtualbmc/tasks/main.yml [hunk 3]
```diff
@@ -18,7 +18,7 @@
   ansible.builtin.file:
     path: "{{ item.key }}"
     state: directory
-    mode: "{{ item.mode | default(omit) }}"
+    mode: "{{ item.mode | default("0755") }}"
```
Comment (lewisdenny) [inline roles/virtualbmc/tasks/main.yml:21]:
```suggestion
    mode: "{{ item.mode | default('0755') }}"
```

Need to use single quotes inside double quotes

Comment (lewisdenny) [reply roles/virtualbmc/tasks/main.yml:21]: That should fix the pre-commit failure

Comment (lewisdenny) [reply roles/virtualbmc/tasks/main.yml:21]: and the virtualbmc failure as well :)

**Review by lewisdenny — CHANGES_REQUESTED** [review]
`cifmw-molecule-install_openstack_ca`  is flaky, not a failure due to this patch, I opened a patch to try improve [here](https://github.com/openstack-k8s-operators/ci-framework/pull/2155)

The other two failures are an easy fix, I left a suggestion. 

This patch is almost ready to get in.

**Review by lewisdenny — CHANGES_REQUESTED** [review]
Please also remove this [1] so we can see if the linter passes with your changes :) 

[1] https://github.com/openstack-k8s-operators/ci-framework/blob/468cdb3f73d8a828b34dabd8e1972193dbf61743/.ansible-lint#L45

File: roles/virtualbmc/tasks/main.yml (hunk 3 above)
Comment (frenzyfriday) [reply roles/virtualbmc/tasks/main.yml:21]: Thanks 🤦🏽‍♀️  I missed this

**Comment by frenzyfriday** [comment]
recheck

**Comment by frenzyfriday** [comment]
recheck

**Comment by lewisdenny** [comment]
Okay, `cifmw-molecule-update_containers` is a real issue.

File: roles/update_containers/tasks/main.yml [hunk 4]
```diff
@@ -24,6 +24,7 @@
   ansible.builtin.template:
     src: "update_containers.j2"
     dest: "{{ cifmw_update_containers_dest_path }}"
+    mode: "0755"
```
Comment (marios) [inline roles/update_containers/tasks/main.yml:27]: is that one right? this looks like a file not a directory and you have used 0644 for all other files

File: roles/tempest/tasks/create-clouds-file.yml [hunk 5]
```diff
@@ -36,3 +36,4 @@
   ansible.builtin.template:
     src: clouds.yaml.j2
     dest: "{{ cifmw_tempest_artifacts_basedir }}/clouds.yaml"
+    mode: "0775"
```
Comment (marios) [inline roles/tempest/tasks/create-clouds-file.yml:39]: should it be 0644 like all other files here?

File: roles/run_hook/molecule/default/prepare.yml [hunk 6]
```diff
@@ -33,6 +34,7 @@
       ansible.builtin.template:
         dest: "/tmp/{{ item }}"
         src: "dummy.yml.j2"
+        mode: "0775"
```
Comment (marios) [inline roles/run_hook/molecule/default/prepare.yml:37]: why is this one 0775 what is the difference from the task above in this file which is 0644

File: roles/reproducer/tasks/ci_job.yml [hunk 7]
```diff
@@ -35,20 +35,23 @@
           ansible.builtin.copy:
             src: "{{ _reproducer_basedir }}/parameters/"
             dest: "/home/zuul/{{ job_id }}-params"
+            mode: "0775"
```
Comment (marios) [inline roles/reproducer/tasks/ci_job.yml:38]: i think all the tasks in this ci_job.yml are for files not directories so is the 0775 intentional here?

**Review by marios — CHANGES_REQUESTED** [review]
sanity check on the permissions for some of the files changes

File: roles/tempest/tasks/create-clouds-file.yml (hunk 5 above)
Comment (pablintino) [reply roles/tempest/tasks/create-clouds-file.yml:39]: +1

File: roles/tofu/tasks/remote_prepare.yml [hunk 8]
```diff
@@ -18,9 +18,11 @@
   ansible.builtin.file:
     path: '{{ cifmw_tofu_dir_path }}'
     state: directory
+    mode: "0755"
 
 - name: "Copy Local Tufo Project Directory To Remote Host '{{ ansible_host }}'"
   ansible.builtin.copy:
+    mode: "0644"
```
Comment (lewisdenny) [inline roles/tofu/tasks/remote_prepare.yml:25]: As `cifmw_tofu_project_path` is an unvalidated user input the user could provide an input that doesn't end with `/` meaning the path would be recursively copied. [1] 

We should also add `directory_mode: "0755"` to cover for this to be explicit with our copied directory modes as well.[2]

```suggestion
    mode: "0644"
    directory_mode: "0755"
```

[1] https://docs.ansible.com/ansible/latest/collections/ansible/builtin/copy_module.html#parameter-src
[2] https://docs.ansible.com/ansible/latest/collections/ansible/builtin/copy_module.html#parameter-directory_mode

**Review by lewisdenny — CHANGES_REQUESTED** [review]
Hi Ananya, 
Please review these suggested changes, I believe this one is close to merging.

File: roles/reproducer/tasks/ci_job.yml (hunk 7 above)
Comment (lewisdenny) [reply roles/reproducer/tasks/ci_job.yml]: +1

File: roles/build_containers/molecule/hotfix/converge.yml [hunk 9]
```diff
@@ -35,6 +35,7 @@
       ansible.builtin.file:
         path: /tmp/rpms
         state: directory
+        mode: "0775"
```
Comment (marios) [inline roles/build_containers/molecule/hotfix/converge.yml:38]: some directories are marked 0755 while others are marked 0775 

is that intentional?

**Review by marios — CHANGES_REQUESTED** [review]
another sanity check please 0775 vs 0755

File: tests/integration/targets/make/tasks/ci_make.yml [hunk 10]
```diff
@@ -32,6 +34,7 @@
     dest: /tmp/artifacts/foobar
     content: |
       I'm a useless flag in here
+    mode: "0755"
```
Comment (marios) [inline tests/integration/targets/make/tasks/ci_make.yml:37]: this looks like a file so 0644

**Review by marios — CHANGES_REQUESTED** [review]
could only find one this time :)
and directories now look consistent at 0755 I couldn't find any more 0775

File: roles/build_containers/molecule/hotfix/converge.yml (hunk 9 above)
Comment (marios) [reply roles/build_containers/molecule/hotfix/converge.yml]: this was updated so now we only have 0755
marking this one resolved thanks

**Comment by frenzyfriday** [comment]
recheck

**Comment by rlandy** [comment]
recheck

**Review by marios — APPROVED** [review]

**Comment by raukadah** [comment]
recheck after dequeuing stuck zuul jobs

**Pull request merged**